src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FinalLocalVariableCheck.java

```diff
@@ -667,6 +697,9 @@ private static boolean isLoopAst(int ast) {
         /** Contains definitions of uninitialized variables. */
         private final Deque<DetailAST> uninitializedVariables = new ArrayDeque<>();
 
+        /** Whether there is a {@code break} or {@code return} in the scope. */
+        private boolean containsBreakOrReturn;
```

Comment:
> I am also going to assume that we don't have any handling of `continue`

You are correct. `continue` should be taken into consideration as well.

@rnveach Regrading the `break` in loops, please point out if I am wrong...
1. In most cases, `break` or `return` does not have a direct loop parent, they are always wrapped by a `if` scope (or `else` scope). In fact, a `break` or `return` with a direct loop parent (smth like below) is really weird.
```
while (...) {
  // statements...
  break;
}
```
In that case we will be sure to break at the first time. If so, why should we use a loop at all? There is no regression problem due to the "`break` in loop" in diff report, I guess it is because no one wrote a `break` or `return` like this in the projects.
2. Actually we don't care about `break` or `return` exit what scope.
```
int a;
if (...) {
  a = 1; // (0)
  break; // or return, we don't care that
}
a = 2; // (1)
```
The `break` or `return` issue is that we ignore them before and regard them as common statements. And the check will consider that `a` is possible to be assigned twice at (0) and (1). And what we should do is to let the check know if there is a `break` or `return` in the `if` scope, then it is not possible to reach both (0) and (1). 
In the example I don't write the enclosing scope at all, it could be a `switch`, and also `for`, `while` etc. It doesn't matter.
3. One thing we need confirm: should we concern about unreachable statement?
> `return` will exit the method completely and ignore any logic outside scope.

It is true but `return` can only be at the end of a `if`, `else` scope (or end of a `case` group, or the end of a method). If there are anything else after the `return`, like smth below:
```
void foo() {
  return;
  final int a = 1;
}
```
We will get a compile error.

\-\-\-
To sum up, IMO, we don't need to handle `break` and `return` in a different way. Please provide a counterexample if I am wrong.
The only case I could find it different now is `break` with a label:
```
int a;
// ...
label: {
  a = 1;
}
// ...
break label;
```
I haven't try to handle with that yet. It seems really hard to handle with.

> Can you create a some more examples with just `break` and `return` by itself

Sure. I will make new comment when I complete sooner